util/agent/agent.go

```diff
@@ -0,0 +1,75 @@
+package agent
+
+import (
+	"context"
+	"errors"
+)
+
+// Agent guards asynchronous changes of an individual location.
+// Inspired by <https://clojure.org/reference/agents>.
+// Note: This is not quite a Clojure agent, since there is no way to deref.
+type Agent struct {
```

Comment:
I've written something like this about ~20 times in Go-world, and it's always subtle and difficult to get right. This is my only ever successful attempt generalizing it to something reusable. Not 100% satisfied with it, but at least it seems correct in isolation.